LearningJourney/Modules/Features/JLibrary/JLibraryTests/Sources/Data/LibraryParserTests.swift

```diff
@@ -0,0 +1,50 @@
+import XCTest
+@testable import JLibrary
+
+final class LibraryParserTests: XCTestCase {
+    // MARK: - System under test
+
+    private let sut = LibraryParser()
+
+    // MARK: - Tests
+
+    func testParseDecodableDataShouldSucceed() throws {
+        // Given
+        let decodableData = DecodableData()
+        let encodedData = try JSONEncoder().encode(decodableData)
+
+        // When
+        let parseResult: Result<DecodableData, ParsingError> = sut.parse(encodedData)
+
+        // Then
+        switch parseResult {
+            case .success(let data):
+                XCTAssertEqual(data, decodableData)
+            case .failure(let error):
+                XCTFail("Expected decoded data but got \(error)")
+        }
+    }
```

Comment:
This test is bloated, coupled with JSONEncoder and complex
```suggestion
    func testParseDecodableDataShouldSucceed() throws {
        // Given
        let decodable = "aDecodableString"
        let encodedData = try XCTUnwrap(decodable.data(encoding: .utf8))
        let expectedDecodedData: Result<String, Error> = .success(decodable)

        // When
        let actualParsedValue = sut.parse(encodedData)

        // Then
        
        XCTAssertEqual(actualParsedValue, expectedDecodedData)
    }
```